iothub/service/src/Messaging/MessagesClient.cs

```diff
@@ -38,6 +38,21 @@ protected MessagesClient()
         {
         }
 
+        /// <summary>
+        /// Creates an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        internal MessagesClient(
```

Comment:
```suggestion
        protected internal MessagesClient(
```

Reply:
This is not possible as parameter types are less accessible than the method